_inc/lib/debugger/class-jetpack-cxn-test-base.php

```diff
@@ -49,21 +51,91 @@ public function __construct() {
 	/**
 	 * Adds a new test to the Jetpack Connection Testing suite.
 	 *
+	 * @since 7.1.0
+	 * @since 7.3.0 Adds name parameter and returns WP_Error on failure.
+	 *
 	 * @param callable $callable Test to add to queue.
-	 * @param array    $groups Testing groups to add test to.
+	 * @param string   $name Unique name for the test.
+	 * @param string   $type   Optional. Core Site Health type: 'direct' if test can be run during initial load or 'async' if test should run async.
+	 * @param array    $groups Optional. Testing groups to add test to.
 	 *
-	 * @return bool True if successfully added. False for a failure.
+	 * @return mixed True if successfully added. WP_Error on failure.
 	 */
-	public function add_test( $callable, $groups = array( 'default' ) ) {
+	public function add_test( $callable, $name, $type = 'direct', $groups = array( 'default' ) ) {
+		if ( is_array( $name ) ) {
+			// Pre-7.3.0 method passed the $groups parameter here.
+			return new WP_Error( __( 'add_test arguments changed in 7.3.0. Please reference inline documentation.', 'jetpack' ) );
+		}
+		if ( array_key_exists( $name, $this->tests ) ) {
+			return new WP_Error( __( 'Test names must be unique.', 'jetpack' ) );
+		}
 		if ( is_callable( $callable ) ) {
-			$this->tests[] = array(
+			$this->tests[ $name ] = array(
 				'test'  => $callable,
 				'group' => $groups,
+				'type'  => $type,
 			);
 			return true;
 		}
 
-		return false;
+		return new WP_Error( __( 'Tests must be valid PHP callables.', 'jetpack' ) );
+	}
+
+	/**
+	 * Lists all tests to run.
+	 *
+	 * @since 7.3.0
+	 *
+	 * @param string $type Optional. Core Site Health type: 'direct' or 'async'. All by default.
+	 * @param string $group Optional. A specific testing group. All by default.
+	 *
+	 * @return array $tests Array of tests with test information.
+	 */
+	public function list_tests( $type = 'all', $group = 'all' ) {
+		if ( ! ( 'all' === $type || 'direct' === $type || 'async' === $type ) ) {
+			_doing_it_wrong( 'Jetpack_Cxn_Test_Base->list_tests', 'Type must be all, direct, or async', '7.3.0' );
+		}
+
+		$tests = array();
+		foreach ( $this->tests as $name => $value ) {
+			// Get all valid tests by group staged.
+			if ( 'all' === $group ) {
+				$tests[ $name ] = $value;
+			} elseif ( $group === $value['group'] ) {
+				$tests[ $name ] = $value;
+			}
```

Comment:
Can we combine these two conditions, since the assignment is identical?

Reply:
d4ded32dd